Sources/Sentry/Public/SentrySpan.h

```diff
@@ -1,9 +1,93 @@
-#import <Foundation/Foundation.h>
+#import "SentrySpanContext.h"
+#import <Sentry/Sentry.h>
 
 NS_ASSUME_NONNULL_BEGIN
 
+@class SentryTransaction, SentrySpanId, SentryId, SentryHub;
+
 NS_SWIFT_NAME(Span)
-@protocol SentrySpan <NSObject>
+@interface SentrySpan : SentrySpanContext
+SENTRY_NO_INIT
+
+/**
+ * The timestamp of which the span ended.
+ */
+@property (nullable, nonatomic, strong) NSDate *timestamp;
+
+/**
+ * The start time of the span.
+ */
+@property (nullable, nonatomic, strong) NSDate *startTimestamp;
+
+/**
+ * An arbitrary mapping of additional metadata of the span.
+ */
+@property (nullable, readonly) NSDictionary<NSString *, id> *extras;
+
+/**
+ * Whether the span is finished.
+ */
+@property (readonly) bool isFinished;
+
+/**
+ * Init a SentrySpan with given transaction, traceId, parentSpanId and hub.
+ *
+ * @param transaction The transaction associated with this span.
+ * @param traceId Determines which trace the Span belongs to.
+ * @param parentId Id of a parent span.
+ *
+ * @return SentrySpan
+ */
+- (instancetype)initWithTransaction:(SentryTransaction *)transaction
+                            traceId:(SentryId *)traceId
+                        andParentId:(SentrySpanId *)parentId;
```

Comment:
`h`: Is this a typical Objective-C naming pattern? We don't use it much in our codebase. I think we only have it in the `SentryHub`. I see that the whole performance API has it now. I rather stick to not prefix the last argument with `and`, unless we have good reasons for that.
```suggestion
                        parentId:(SentrySpanId *)parentId;
```

Reply:
I think I'm abusing the "and" use.
Apple does use it as described: "If the method describes two separate actions, use “and” to link them."
Old habits. Let's not use it.

Reply:
Yes, but in this case, it's not two separate actions. Thanks for getting rid of it. Please check the whole performance API because I think it is in several places.